### PR TITLE
[macOS] CoreRT native compilation for select projects targeting osx

### PR DESCRIPTION
**Pull request by almiliMSFT**
Some SDKs for CoreRT native compilation on macOS.

For now, only `AppHostPatcher` is compiled to native (and that executable is subsequently used to patch binaries throughout the rest of the build).

Future PRs can enable compiling other executables to native.

### CODE REVIEW

File: Public/Sdk/Public/Managed/Tools/AppHostPatcher/AppHostPatcher.dsc [hunk 1]
```diff
@@ -3,33 +3,31 @@
 
 import {Artifact, Cmd, Transformer} from "Sdk.Transformers";
 import * as Managed from "Sdk.Managed";
+import * as Shared from "Sdk.Managed.Shared";
```
Comment (Kr1sso) [inline Public/Sdk/Public/Managed/Tools/AppHostPatcher/AppHostPatcher.dsc]: Nit: Maybe order by Sdk.Manager.*?

File: Public/Sdk/Public/Managed/Tools/AppHostPatcher/AppHostPatcher.dsc [hunk 2]
```diff
@@ -3,33 +3,31 @@
...
-        ...addIfLazy(Context.getCurrentHost().os === "macOS", () => [
-            ...MacOS.untrackedSystemFolderDeps,
-            d`${Context.getMount("UserProfile").path}/Microsoft/BuildXL/RestrictedTemp`,
-        ])
-    ],
-    untrackedFiles: [
-        ...addIfLazy(Context.getCurrentHost().os === "macOS", () => MacOS.untrackedFiles)
-    ],
+    exe: patcherExecutable,
+    dependsOnCurrentHostOSDirectories: true,
```
Comment (Kr1sso) [inline Public/Sdk/Public/Managed/Tools/AppHostPatcher/AppHostPatcher.dsc]: I assume this means things like `untrackedSystemFolderDeps` are now automatically added in our backend code, we have a lot of other `untrackedDirectoryScopes` in other tools, do you mind making a pass over that and removing explicit calls and replacing them with `dependsOnCurrentHostOSDirectories` if that is the cleaner way?

File: Public/Sdk/Public/Managed/Tools/ILCompiler/ilc.dsc [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    emitDebugInformation: true,
+    completeTypeMetadata: true,
+    rootAllApplicationAssemblies: true,
+    dependencies: [
+        pkgContents
+    ]
+};
+
+@@public
+export interface Arguments extends Transformer.RunnerArguments {
```
Comment (Kr1sso) [inline Public/Sdk/Public/Managed/Tools/ILCompiler/ilc.dsc]: If there is an overview somewhere a comment with a link to what does options mean would be helpful.

Comment (almiliMSFT) [reply Public/Sdk/Public/Managed/Tools/ILCompiler/ilc.dsc]: I copy pasted option descriptions from `ilc --help`

**Comment by almiliMSFT** [comment]
SDKs for CoreRT native compilation on macOS.

For now, only AppHostPatcher is compiled to native (and that executable is subsequently used to patch binaries throughout the rest of the build).

Future PRs can enable compiling other executables to native.

**Pull request merged**